app/models/order_detail.rb

```diff
@@ -942,7 +942,7 @@ def clear_statement
 
   def update_fulfilled_at_on_resolve
     if problem_changed? && !problem_order?
-      self.fulfilled_at = time_data.actual_end_at
+      self.fulfilled_at = time_data.actual_end_at if time_data.actual_end_at
```

Comment:
Would doing `self.fulfilled_at ||= time_data.actual_end_at` make sense? There's two scenarios to think about:
1) Resolving missing actual time (I believe that's what this is ultimately trying to handle)
2) Resolving missing price policies (it probably shouldn't change much in that case)